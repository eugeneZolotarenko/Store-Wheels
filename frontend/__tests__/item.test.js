import ItemComponent from "../components/Item";
import UserComponent from "../components/User";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: "ABCD123",
  title: "An Item",
  price: 5000,
  description: "This is cool item",
  image: "dog.jpg",
  largeImage: "largedog.jpg"
};

describe("<Item/>", () => {
  it("renders and matches the snapshot", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it("renders the image properly", () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find("img");
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });
  // it("renders PriceTag and title properly", () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find("PriceTag");
  //   console.log(PriceTag.debug());
  //   expect(PriceTag.children().text()).toBe("$50");
  //   expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  // });
});
