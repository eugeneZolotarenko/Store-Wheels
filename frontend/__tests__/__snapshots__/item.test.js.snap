// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <Link
    href={
      Object {
        "pathname": "/item",
        "query": Object {
          "id": "ABCD123",
        },
      }
    }
  >
    <img
      alt="An Item"
      src="dog.jpg"
    />
  </Link>
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABCD123",
          },
        }
      }
    >
      <a>
        An Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This is cool item
  </p>
  <ButtonList
    item={
      Object {
        "description": "This is cool item",
        "id": "ABCD123",
        "image": "dog.jpg",
        "largeImage": "largedog.jpg",
        "price": 5000,
        "title": "An Item",
      }
    }
  />
</ItemStyles__Item>
`;
